integration/src/test/java/org/mule/test/integration/transaction/TransactionRolledBackByOwnerTestCase.java

```diff
@@ -247,11 +249,20 @@ public String describeFailure() {
   private void assertCorrectStates() {
     if (ignoreExtraStates) {
       for (int i = 0; i < expectedStates.size(); i++) {
-        assertThat(states.get(i), is(expectedStates.get(i)));
+        assertThat("Expected " + expectedStates + " but obtained " + states, containsSequence(states, expectedStates), is(true));
```

Comment:
It doesn't look like this should be run inside a loop anymore.

Reply:
Absolutely not :piccard-facepalm:.